tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+                    )
+                    assert len(filtered_stacks) == 1
+
+
+# .-----------.
+# | Pipelines |
+# '-----------'
+
+# .----------------.
+# | Pipeline runs  |
+# '----------------'
+
+
+def test_list_runs_is_ordered():
+    """Tests listing runs returns ordered runs."""
+    client = Client()
+    store = client.zen_store
+
+    num_pipelines_before = store.list_runs(PipelineRunFilterModel()).total
+
+    num_runs = 5
+    with PipelineRunContext(num_runs):
+
+        pipelines = store.list_runs(PipelineRunFilterModel()).items
+        assert len(pipelines) == num_pipelines_before + num_runs
+        assert all(
+            pipelines[i].created <= pipelines[i + 1].created
+            for i in range(len(pipelines) - 1)
+        )
+
+
+def test_deleting_run_deletes_steps():
+    """Tests deleting run deletes its steps."""
+    client = Client()
+    store = client.zen_store
+
+    # Just in case the test db is not in a clean state we compare relative
+    num_steps_before = store.list_run_steps(StepRunFilterModel()).total
+    num_pipelines_before = store.list_runs(PipelineRunFilterModel()).total
+
+    num_runs = 1
+
+    with PipelineRunContext(num_runs) as runs:
+        steps = store.list_run_steps(StepRunFilterModel())
+
+        assert steps.total == num_steps_before + num_runs * 2
+        pipelines = store.list_runs(PipelineRunFilterModel())
+        assert pipelines.total == num_pipelines_before + num_runs
+        run_id = runs[0].id
+        store.delete_run(run_id)
+        assert (
+            len(store.list_run_steps(StepRunFilterModel())) == num_steps_before
+        )
+
+
+# .--------------------.
+# | Pipeline run steps |
+# '--------------------'
+
+
+def test_get_run_step_outputs_succeeds():
+    """Tests getting run step outputs."""
+    client = Client()
+    store = client.zen_store
+
+    with PipelineRunContext(1):
+        steps = store.list_run_steps(StepRunFilterModel(name="step_2"))
+
+        for step in steps.items:
+            run_step_outputs = store.get_run_step(step.id).output_artifacts
+            assert len(run_step_outputs) == 1
+
+
+def test_get_run_step_inputs_succeeds():
+    """Tests getting run step inputs."""
+    client = Client()
+    store = client.zen_store
+
+    with PipelineRunContext(1):
+
+        steps = store.list_run_steps(StepRunFilterModel(name="step_2"))
+        for step in steps.items:
+            run_step_inputs = store.get_run_step(step.id).input_artifacts
+            assert len(run_step_inputs) == 1
+
+
+# .-----------.
+# | Artifacts |
+# '-----------'
+
+
+def test_list_unused_artifacts():
+    """Tests listing with `unused=True` only returns unused artifacts."""
+    client = Client()
+    store = client.zen_store
+
+    num_artifacts_before = store.list_artifacts(ArtifactFilterModel()).total
+    num_unused_artifacts_before = store.list_artifacts(
+        ArtifactFilterModel(only_unused=True)
+    ).total
+    num_runs = 1
```

Comment:
Same here, this will fail if run in parallel (local machine or different machines)